libr/core/graph.c

```diff
@@ -2218,6 +2218,28 @@ R_API RAGraph *r_agraph_new(RConsCanvas *can) {
 	return g;
 }
 
+static void visual_offset (RCore *core) {
+	char buf[256];
+	r_line_set_prompt ("[offset]> ");
+	strcpy (buf, "s ");
+	if (r_cons_fgets (buf+2, sizeof (buf)-3, 0, NULL) >0) {
+		if (buf[2]=='.')buf[1]='.';
+		r_core_cmd0 (core, buf);
+	}
+}
+
+static void print_agraph (RAGraph *g, RCore *core, struct agraph_refresh_data *grd){
+	RAnalFunction **fcn = grd->fcn;
+	RAnalFunction *f = r_anal_get_fcn_in (core->anal, core->offset, 0);
+	if (f && f != *fcn){
+		*fcn = f;
+		g->need_reload_nodes = R_TRUE;
+	}
+	agraph_print (g, grd->fs, core, *fcn);
```

Comment:
The graph is always re-printed even when not in debug mode. That comment was there because so far changing function wasn't supported. I think now we can remove it and extract the part similar to your function outside the if
```
 if (g->is_instep && core->io->debug)
    r_core_cmd0 (core, "sr pc");
RAnalFunction *f = r_anal_get_fcn_in (core->anal, core->offset, 0);
if (f && f != *fcn) {
            *fcn = f;
            g->need_reload_nodes = R_TRUE;
}
```
We avoid duplicated code in this way